src/PolicyStoreImp.cpp

```diff
@@ -300,6 +300,13 @@ namespace geopm
                                   const std::string &profile_name,
                                   const std::vector<double> &policy)
     {
+        // check that the policy is valid for the agent
+        size_t num_policy = Agent::num_policy(agent_name);
+        if (policy.size() > num_policy) {
```

Comment:
Is it possible that policy.size() would be < num_policy?  Is that also an error condition?

Reply:
This is allowed, and the missing policy values will be omitted from the policystore.  There is a test of this behavior on line 145 of PolicyStoreTest.cpp